### ISSUE TITLE
Custom "attribute_set_id" filter applied by the collection processors override condition type in search criteria

### ISSUE THREAD

**Issue by Serializator** [magento/magento2#29403]
### Preconditions (*)
1. Magento 2.4 / 2.3.5-p2 / 2.4-develop

### Steps to reproduce (*)
1. Create one or more attribute sets
2. Create one or more attribute groups in these attribute sets
3. Try to retrieve these attribute groups from both attribute sets using the REST endpoint below

`/rest/V1/products/attribute-sets/{attribute-set-id}/groups/list?searchCriteria[filter_groups][0][filters][0][field]=attribute_set_id&searchCriteria[filter_groups][0][filters][0][value]={attribute_set_ids_comma_separated}&searchCriteria[filter_groups][0][filters][0][condition_type]=in`

### Expected result (*)
1. Attribute groups from the given attribute sets in the search criteria

### Actual result (*)
1. No results or only the results from the attribute set of which the id is the first id in the comma separated list of ids ("14,25,30,60" would only return the attribute groups from the attribute set with id 14, for example)

### Additional
`Magento\Eav\Model\Api\SearchCriteria\CollectionProcessor\FilterProcessor\AttributeGroupAttributeSetIdFilter` does not take into account any of the other condition types supported by the search criteria as described in the Magento API documentation.

https://devdocs.magento.com/guides/v2.4/rest/performing-searches.html

It overrides any given condition type with `eq`.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### Additional info from Engcom
Attribute set id from `eav_attribute_group` table 
![at_set_database](https://user-images.githubusercontent.com/51680745/106254250-2d657a00-6221-11eb-8832-106688527aea.png)
GET catalogProductAttributeGroupRepositoryV1
![execute_query](https://user-images.githubusercontent.com/51680745/106254386-50902980-6221-11eb-9e82-76c503835fa1.png)

The response body contains only 1, despite that  three values are entered comma separated 
![response_body](https://user-images.githubusercontent.com/51680745/106254724-bda3bf00-6221-11eb-9ba4-5a50ae746cdc.png)

**Comment by Serializator** [comment]
@magento-engcom-team, I do not believe this issue should be closed because it is "stale", as such a comment so that it won't be closed. I think all information necessary is in the description of the issue. Please let me know if you need any more information!

**Comment by engcom-Bravo** [comment]
Hello @Serializator 

Thank you for your report.

We Confirm that the issue is reproducible on the latest 2.4-develop. Additional info is added to the Description.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-40676` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by Serializator** [comment]
@magento-engcom-team, I do not believe this issue should be closed because it is "stale", as such a comment so that it won't be closed. I think all information necessary is in the description of the issue. Please let me know if you need any more information!

**Comment by makzef** [comment]
@magento I am working on this
